Editor WOH (org.opencastproject.videoeditor): line did not start with WEBVTT. Got ﻿WEBVTT

<!-- Delete/replace placeholder texts -->

**Describe the bug**

When attempting to cut a subtitle with Byte-Order-Mark (BOM), the workflow may fail.

**To Reproduce**

Steps to reproduce the behavior:

I'll come back to this later today. TLDR; add WEBVTT subtitles as a track and configure the editor step to cut them.

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error Exception message ProcessFailedException: Unable to process SMIL: Header line did not start with WEBVTT. Got ﻿WEBVTT


**Expected behavior**

Can cut WEBVTT files with BOM.

According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/WebVTT_API/Web_Video_Text_Tracks_Format):

> The structure of a WebVTT consists of the following components, some of them optional, in this order:
> * A header, consisting of an optional byte **order mark (BOM)** — the string `WEBVTT` — followed by an optional text header separated by one or more space or tab characters (in WebVTT files, tabs and spaces are interchangeable).

**Stack trace**

If applicable, add a stack trace to help explain your problem.
```
org.opencastproject.videoeditor.api.ProcessFailedException: Unable to process SMIL: Header line did not start with WEBVTT. Got ﻿WEBVTT
	at org.opencastproject.videoeditor.impl.VideoEditorServiceImpl.processSmil(VideoEditorServiceImpl.java:432)
	at org.opencastproject.videoeditor.impl.VideoEditorServiceImpl.process(VideoEditorServiceImpl.java:530)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313)
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.opencastproject.subtitleparser.SubtitleParsingException: Header line did not start with WEBVTT. Got ﻿WEBVTT
	at org.opencastproject.subtitleparser.webvttparser.WebVTTParser.parse(WebVTTParser.java:81)
	at org.opencastproject.videoeditor.impl.VideoEditorServiceImpl.processSmil(VideoEditorServiceImpl.java:356)
	... 7 more
```

The error might be https://github.com/opencast/opencast/blob/develop/modules/subtitle-parser/src/main/java/org/opencastproject/subtitleparser/webvttparser/WebVTTParser.java#L80

```java
    if (!line.startsWith("WEBVTT")) {
      throw new SubtitleParsingException("Header line did not start with WEBVTT. Got " + line);
    }
```

**Server environment:**
 - Opencast Container Images; Opencast 14.13


**Additional context**

Add any other context about the problem here.

--- a/modules/subtitle-parser/pom.xml
+++ b/modules/subtitle-parser/pom.xml
@@ -18,6 +18,10 @@
 
   <dependencies>
     <dependency>
+      <groupId>commons-io</groupId>
+      <artifactId>commons-io</artifactId>
+    </dependency>
+    <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <scope>test</scope>

--- a/modules/subtitle-parser/src/main/java/org/opencastproject/subtitleparser/webvttparser/WebVTTParser.java
+++ b/modules/subtitle-parser/src/main/java/org/opencastproject/subtitleparser/webvttparser/WebVTTParser.java
@@ -22,6 +22,8 @@ package org.opencastproject.subtitleparser.webvttparser;
 
 import org.opencastproject.subtitleparser.SubtitleParsingException;
 
+import org.apache.commons.io.input.BOMInputStream;
+
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
@@ -64,11 +66,14 @@ public class WebVTTParser {
   }
 
   public WebVTTSubtitle parse(InputStream is) throws IOException, SubtitleParsingException {
+    // Wrap input stream into Apache Commons IO BOMInputStream
+    BOMInputStream bomIn = new BOMInputStream(is, false);
+
     // Create subtitle object
     WebVTTSubtitle subtitle = new WebVTTSubtitle();
 
     // Read each line
-    BufferedReader webvttReader = new BufferedReader(new InputStreamReader(is, this.charset));
+    BufferedReader webvttReader = new BufferedReader(new InputStreamReader(bomIn, this.charset));
     String line = "";
 
     // File should start with "WEBVTT" on the first line

--- a/modules/subtitle-parser/src/test/java/org/opencastproject/subtitleparser/webvttparser/WebVTTTest.java
+++ b/modules/subtitle-parser/src/test/java/org/opencastproject/subtitleparser/webvttparser/WebVTTTest.java
@@ -38,10 +38,12 @@ import java.net.URISyntaxException;
  */
 public class WebVTTTest {
   protected String inputFilePath;
+  protected String inputFileWithBom;
   protected String outputFilePath;
 
   public WebVTTTest() throws URISyntaxException {
     inputFilePath = new File(getClass().getResource("/testresources/example.vtt").toURI()).getAbsolutePath();
+    inputFileWithBom = new File(getClass().getResource("/testresources/example-bom.vtt").toURI()).getAbsolutePath();
     outputFilePath = new File("target/testoutput/output.vtt").getAbsolutePath();
   }
 
@@ -61,9 +63,23 @@ public class WebVTTTest {
   }
 
   @Test
+  public void parseBomWithoutException() throws IOException, SubtitleParsingException {
+    WebVTTParser parser = new WebVTTParser();
+    parser.parse(new FileInputStream(inputFileWithBom));
+  }
+
+  @Test
   public void parseCorrectly() throws IOException, SubtitleParsingException {
     WebVTTParser parser = new WebVTTParser();
     WebVTTSubtitle subtitle = parser.parse(new FileInputStream(inputFilePath));
+
+    assertExampleVtt(subtitle);
+  }
+
+  @Test
+  public void parseBomCorrectly() throws IOException, SubtitleParsingException {
+    WebVTTParser parser = new WebVTTParser();
+    WebVTTSubtitle subtitle = parser.parse(new FileInputStream(inputFileWithBom));
 
     assertExampleVtt(subtitle);
   }

--- a/modules/subtitle-parser/src/test/resources/testresources/example-bom.vtt
+++ b/modules/subtitle-parser/src/test/resources/testresources/example-bom.vtt
@@ -1,0 +1,30 @@
+﻿WEBVTT There can be additional characters here
+kind: the best kind
+
+REGION
+There could be some region text here
+
+STYLE
+::cue(b) {
+  color: peachpuff;
+}
+
+REGION
+There could be some region text here
+So much region text
+
+x-o7MdlMG-qDMsG6oWbDm
+00:00.000 --> 00:05.000 vertical:undefined align:undefined size:undefined% line:undefined% position:undefined%
+Test Cue 1
+
+00:05.000 --> 00:06.000 vertical:undefined align:undefined size:undefined% line:undefined% position:undefined%
+Test Cue 2
+
+NOTE comment blocks can be used everywhere
+
+00:11.123 --> 00:14.456
+- Never drink liquid nitrogen.
+
+00:15.500 --> 00:19.500
+- It will perforate your stomach.
+- You could die.

